chains/evm/deployment/utils/changesets/changesets.go

```diff
@@ -19,33 +18,40 @@ type NewFromOnChainSequenceParams[IN any, DEP any, CFG any] struct {
 	ResolveInput func(e deployment.Environment, cfg CFG) (IN, error)
 	// ResolveDeps resolves the dependencies for the sequence based on the environment and changeset config.
 	ResolveDep func(e deployment.Environment, cfg CFG) (DEP, error)
+	// ResolveMCMS resolves the MCMS configuration based on the environment and changeset config.
+	ResolveMCMS func(e deployment.Environment, cfg CFG) (MCMSParams, error)
```

Comment:
Not needed here, as much of the ResolveMCMS logic can just be baked into NewFromOnChainSequence.

What we should have here is `ResolveMCMSMetadata(e cldf_deployment.Environment, mcmsRef datastore.AddressRef) (mcms_types.ChainMetadata, error)`, which would get called for each MCM contract used in the proposal. The ref loaded from datastore gets passed in.

Then we would have a resolveMCMSMetadataForEVM that we can pass in